pkg/workload/histogram/exporter/util.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package exporter
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/gogo/protobuf/proto"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+)
+
+func sanitizeOpenmetricsLabels(input string) string {
+	regex := regexp.MustCompile(`[^a-zA-Z0-9_]`)
+	sanitized := regex.ReplaceAllString(input, "_")
+	firstCharRegex := regexp.MustCompile(`^[^a-zA-Z_]`)
+	sanitized = firstCharRegex.ReplaceAllString(sanitized, "_")
+	return sanitized
+}
+
+// ConvertHdrHistogramToPrometheusMetricFamily converts a Hdr histogram into MetricFamily which can be used
+// by expfmt.MetricFamilyToOpenMetrics to export openmetrics
+func ConvertHdrHistogramToPrometheusMetricFamily(
+	h *hdrhistogram.Histogram, name *string, start time.Time, labels *map[string]string,
+) *io_prometheus_client.MetricFamily {
+	hist := &io_prometheus_client.Histogram{}
+
+	bars := h.Distribution()
+	hist.Bucket = make([]*io_prometheus_client.Bucket, 0, len(bars))
+
+	var cumCount uint64
+	var sum float64
+	for _, bar := range bars {
+		if bar.Count == 0 {
```

Comment:
We set the size of hist.Bucket to len(bars), but then here we skip bar which has bar.Count == 0.
Could this cause any problem?
Why are we skipping these values with bar.Count == 0? a zero value in histogram bucket should also be recorded right?

Reply:
Skipping the ranges or a zero count bucket is ideally same in openmetrics. So if we expose the zero count bucket, we are just filling our storage/file which empty buckets.